### PR TITLE
Add validator transitioner tests

### PR DESCRIPTION
**Pull request by Ullaakut**
## Goal of this PR

This PR brings to master one commit that got lost on the now unused [`validators-ftso-transition`](https://github.com/flare-foundation/coreth/tree/validators-ftso-transition) branch.

PRs were based on it, and when it got merged I didn't realize immediately and I merged PR #24 into it, so it never got to master. This PR fixes this mistake.

It includes a single commit, by @emailtovamos

### CODE REVIEW

File: plugin/evm/validators_transitioner_test.go [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+		validatorsTransitioner := NewValidatorsTransitioner(nil, mock)
+
+		got, err := validatorsTransitioner.ByEpoch(epoch)
+		require.NoError(t, err)
+		assert.Equal(t, providers, got)
+	})
+}
+
+func fakeProviders(epoch uint64) map[ids.ShortID]uint64 {
```
Comment (AnNiran) [inline plugin/evm/validators_transitioner_test.go:144]: Inside tests/mocks/validators.go, there is a specific function `func RetrieverMock(numValidators, epochs int) ValidatorsTestData` that creates random validators set with epochs.
You can simply use it by giving `1` as epoch number and then use the 0th position of the map that is received, 
as data material used across these tests.
Or, use the hardcoded validators values inside `ValidatorsData`.
The functionality for adding an additional byte, in order to be 100% sure the data is not the same, can be run on any of the above results, and IMO it should also be inside tests/mocks/validators as a method implemented by `ValidatorsTestData`, as it already implements `Duplicate`.

Comment (Ullaakut) [reply plugin/evm/validators_transitioner_test.go:144]: This is true on your branch, but it has not been merged yet. We can do a round of cleanup when it gets merged, to uniformize the tests and their use of mocked values 👍

Comment (AnNiran) [reply plugin/evm/validators_transitioner_test.go:144]: > This is true on your branch, but it has not been merged yet. We can do a round of cleanup when it gets merged, to uniformize the tests and their use of mocked values +1
Yes

**Review by AnNiran — APPROVED** [review]
